lib/components/MapView.js

```diff
@@ -430,13 +435,14 @@ class MapView extends React.Component {
   }
 
   _onMapReady() {
-    const { region, initialRegion } = this.props;
+    const { region, initialRegion, onMapReady } = this.props;
     if (region) {
       this.map.setNativeProps({ region });
     } else if (initialRegion) {
       this.map.setNativeProps({ region: initialRegion });
     }
     this._updateStyle();
+    if (onMapReady) onMapReady();
     this.setState({ isReady: true });
```

Comment:
I think you should call `onMapReady` when state has changed. So like this:
```
this.setState({ isReady: true }, () => {
  if (onMapReady) onMapReady();
});
```

Reply:
is there a reason to why `this` is not being passed as an argument to `onMapReady()`? 
refs are set in a arbitrary moment during react renders, and devs does not have control when they are set, thus outputting different results in some of initial renders. By passing the `this` as an argument that would be fixed and we would have the map ref exactly when the map is ready every time. 

May I create a PR for this?

Reply:
@betoharres I don't think you need to do it, the function is bound already - https://github.com/react-native-community/react-native-maps/blob/master/lib/components/MapView.js#L539

Reply:
@rborn thanks for the reply. Unfortunately no arguments are being passed on that callback, I edited the file just by modifying that line to `if (onMapReady) onMapReady(this);` and I got the result as I expected.

I've ran into this problem before and tried to workaround, but the results, as I said in my previous comment, are unstable; sometimes the ref is set, sometimes not.
My main goal is to run `mapRef.animateToRegion(userLocation)` when the map is ready, so that's why I'm interested on that change. 

I have a PR ready to send if you want(besides, I'm just adding a word :laughing: )

Reply:
ref should be set, if is not set you are doing something funny :)

Reply:
Thanks for the reply again! I did some testing and you're right! If you only render the MapView component, it does have the ref set on every `onMapReady()` call, but if I add the [useNetIfno()](https://github.com/react-native-community/react-native-netinfo) hook call to my component, the ref starts to return `null` sometimes when `onMapReady` fires. 
If you are interested, I can open a issue and try to push a example project, because maybe something else could be interfering(even though I'm only rendering the Map with `react-navigation`).
I still like the idea of having an event available to set the ref, I'm not sure how it would affect the current code, but it doesn't seems to hurt I guess :sweat_smile: